src/java/org/apache/cassandra/service/pager/PagingState.java

```diff
@@ -113,12 +118,13 @@ public static PagingState deserialize(ByteBuffer bytes, ProtocolVersion protocol
                 if (isModernSerialized(bytes)) return modernDeserialize(bytes, ProtocolVersion.V4);
             }
         }
-        catch (IOException e)
+        catch (Exception e)
         {
+            logger.error("Protocol exception", e);
```

Comment:
It was mostly for debugging purposes. Exception in catch is not something we would expect thus it will rather not pop up in the logs. Anyway, I'll change it to trace